platform/core/src/studies/services/wado/retrieveMetadata.js

```diff
@@ -452,18 +460,101 @@ async function resultDataToStudyMetadata(server, studyInstanceUid, resultData) {
  * @returns {Promise}
  */
 async function RetrieveMetadata(server, studyInstanceUid) {
-  const config = {
+  return (server.enableStudyLazyLoad
+    ? lazyLoadStudyMetadata
+    : loadStudyMetadata)(server, studyInstanceUid);
+}
+
+async function loadStudyMetadata(server, studyInstanceUID) {
+  const dicomWeb = new api.DICOMwebClient({
     url: server.wadoRoot,
     headers: DICOMWeb.getAuthorizationHeader(server),
-  };
-  const dicomWeb = new api.DICOMwebClient(config);
-  const options = {
-    studyInstanceUID: studyInstanceUid,
-  };
+  });
+  return dicomWeb
+    .retrieveStudyMetadata({ studyInstanceUID })
+    .then(result => createStudyFromInstanceList(server, result));
+}
+
+async function lazyLoadStudyMetadata(server, studyInstanceUid) {
+  const seriesIds = await searchStudySeries(server, studyInstanceUid);
+  if (seriesIds) {
+    const dicomWeb = new api.DICOMwebClient({
+      url: server.wadoRoot,
+      headers: DICOMWeb.getAuthorizationHeader(server),
+    });
+    const seriesLoadQueue = makeSeriesLoadQueue(
+      dicomWeb,
+      studyInstanceUid,
+      seriesIds
+    );
+    const firstSeries = await seriesLoadQueue.dequeue();
+    if (firstSeries && firstSeries.instances) {
+      const study = await createStudyFromInstanceList(
+        server,
+        firstSeries.instances
+      );
+      if (study) {
+        setQueueListeners(server, study, seriesLoadQueue);
+        return study;
+      }
+    }
+  }
+  return null;
+}
 
-  return dicomWeb.retrieveStudyMetadata(options).then(result => {
-    return resultDataToStudyMetadata(server, studyInstanceUid, result);
+function setQueueListeners(server, study, seriesLoadQueue) {
+  study.seriesLoadQueue = seriesLoadQueue;
+  // Series Loaded Event
+  seriesLoadQueue.subscribe(seriesLoadQueue.EVENT_SERIES_LOADED, function(
+    series
+  ) {
+    if (series && series.instances) {
+      addInstancesToStudy(server, study, series.instances).then(
+        function() {
+          study.publish(
+            'seriesAdded',
+            study.seriesMap[series.seriesInstanceUID]
+          );
+        },
+        function(error) {
+          log.error(
+            '[retrieveMetadata] failed to add instance to study on async series load',
+            error
+          );
+        }
+      );
+    }
+  });
+}
+
+async function searchStudySeries(server, studyInstanceUID) {
+  const dicomWeb = new api.DICOMwebClient({
+    url: server.qidoRoot,
+    headers: DICOMWeb.getAuthorizationHeader(server),
```

Comment:
This is also using a DICOMWebClient configured like the other two?

Reply:
This is the one using the `qidoRoot` instead of the `wadoRoot`